Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/CommonComponents/LocationDrawingsButton.tsx

```diff
@@ -0,0 +1,163 @@
+                    const option: DropDownOption = {
+                        Label: location.Name,
+                        Callback: () => handleShowDrawingsModal(location),
+                        Disabled: error != "",
+                        ToolTipContent: <p>{error}</p>,
+                        ShowToolTip: error != "",
+                        ToolTipLocation: "left",
+                        Key: i
+                    };
+                    setLocationOptions(prev => [...prev, option]);
+                })
+                .fail(() => { throw new Error() });
+            return handle;
+        }).filter(handle => handle != null);
+
+        Promise.all(handles)
+            .then(() => setPageState('idle'),
+                () => setPageState('error'));
+
+        return () => handles.forEach(handle => () => { if (handle != null && handle?.abort != null) handle.abort() });
+    }, [props.Locations]);
+
+    const handleShowDrawingsModal = (loc) => {
+        if (loc == undefined) return;
+        setSelectedLocation(loc);
+        setShowDrawingsModal(true);
+    };
+
+    return (
+        <div>
+            <LoadingScreen Show={pageState == 'loading' || props.IsLoadingLocations == true} />
+            <ServerErrorIcon Show={pageState == 'error'} Size={40} Label={'A Server Error Occurred. Please Reload the Application.'} />
+            {!multipleLocations
+            ? <>
+            <button
+                className={locationOptions[0]?.Disabled ? "btn btn-primary disabled" : "btn btn-primary"}
+                onClick={() => locationOptions[0]?.Disabled ? null : handleShowDrawingsModal(props.Locations[0])}
+                data-tooltip={"DrawingsModal"}
+                onMouseEnter={() => setHover('drawings')}
+                onMouseLeave={() => setHover('none')}
+                >Open {props.Locations[0]?.Name} Drawings
+            </button>
+            <ToolTip
+                Show={locationOptions[0]?.Disabled && hover === 'drawings'}
+                Theme={'dark'}
+                Position={'top'}
+                Target={"DrawingsModal"}
+                Zindex={9999}
+                    ><p>{CrossMark} {locationOptions[0]?.ToolTipContent}</p>
+            </ToolTip>
+            </>
+            : <BtnDropdown
+                Label={'Open ' + props.Locations[0]?.Name + ' Drawings'}
+                Callback={() => handleShowDrawingsModal(props.Locations[0])}
+                TooltipContent={
+                    <p>{CrossMark} {locationOptions[0]?.Disabled}</p>
+                }
+                ShowToolTip={locationOptions[0]?.ShowToolTip}
+                Disabled={locationOptions[0]?.Disabled}
+                BtnClass={'btn-primary'}
+                Options={locationOptions.slice(1)}
+            />
+            }
+            <Modal
+                Show={showDrawingsModal}
+                Title={'Drawings for ' + selectedLocation?.Name}
+                ShowX={true} Size={'lg'}
```

Comment:
```suggestion
                ShowX={true} Size={'xlg'}
```
lg is too small to show all columns of the drawing table

Reply:
updated in latest commit